File: services/lucy-dotnet/src/Lucy.AuthPayment.Api/Modules/Payment/Models/Donation.cs
namespace lucy_auth_payment.Modules.Payment.Models;

public class Donation
{
    public string DonationId { get; set; } = null!;
    public string FromUserId { get; set; } = null!;
    public string ToUserId { get; set; } = null!;
    public string? RoomId { get; set; }
    public decimal Amount { get; set; }
    public string? MessageText { get; set; }
    public string? FromWalletTransactionId { get; set; }
    public string? ToWalletTransactionId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/payment/entity/Donation.java
package com.lucy.backend.payment.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "Donations")
@Data
public class Donation {
    @Id
    @Column(name = "DonationId", length = 50)
    private String donationId;

    @Column(name = "GiftId", length = 50, nullable = false)
    private String giftId;

    @Column(name = "FromUserId", length = 50, nullable = false)
    private String fromUserId;

    @Column(name = "ToUserId", length = 50, nullable = false)
    private String toUserId;

    @Column(name = "RoomId", length = 50, nullable = false)
    private String roomId;

    @Column(name = "Quantity", nullable = false)
    private Integer quantity = 1;

    @Column(name = "Amount", precision = 12, scale = 2, nullable = false)
    private BigDecimal amount;

    @Column(name = "TotalAmount", precision = 12, scale = 2, nullable = false)
    private BigDecimal totalAmount;

    @Column(name = "WalletTransactionId", length = 50, nullable = false)
    private String walletTransactionId;

    @Column(name = "MessageText", length = 255)
    private String messageText;

    @Column(name = "CreatedAt", updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}
